jenkins/Jenkinsfile.macos

```diff
@@ -96,196 +96,168 @@ pipeline {
       }
     }
 
-    stage("Build Matrix") {
+    stage('Build and Sign') {
 
       environment {
-        ENV = ""
+        AWS_ACCOUNT_ID = '749683154838'
+        KEYCHAIN_PASSPHRASE = credentials('ide-keychain-passphrase')
+        BUILD_AGENT_TEMP = "${env.HOME}/tmp"
+        MACOS_DEVELOPER_CERTIFICATE = credentials('MACOS_DEVELOPER_CERTIFICATE')
+        MACOS_DEVELOPER_CERTIFICATE_KEY = credentials('MACOS_DEVELOPER_CERTIFICATE_KEY')
       }
 
-      matrix {
+      steps {
+        script {
+          ENV = utils.getBuildEnv(!params.DAILY)
+        }
+        script {
+          // Create a keychain to hold the Developer ID certificate for signing
+          sh """
+            if [ ! -f "${BUILD_AGENT_TEMP}/buildagent.keychain" ]; then
+              security create-keychain -p ${KEYCHAIN_PASSPHRASE} ${BUILD_AGENT_TEMP}/buildagent.keychain
+            else
+              echo "Using keychain: ${BUILD_AGENT_TEMP}/buildagent.keychain"
+            fi
+          """
+        }
+        sh 'security default-keychain -s ${BUILD_AGENT_TEMP}/buildagent.keychain'
+        sh 'security unlock-keychain -p ${KEYCHAIN_PASSPHRASE} ${BUILD_AGENT_TEMP}/buildagent.keychain && security set-keychain-settings' // turn off timeout
+        // Import the Developer ID certificate from Jenkins Secrets.
+        sh 'security import \"${MACOS_DEVELOPER_CERTIFICATE}\" -k ${BUILD_AGENT_TEMP}/buildagent.keychain -P \"${MACOS_DEVELOPER_CERTIFICATE_KEY}\" -T /usr/bin/codesign'
```

Comment:
Are the escapes here necessary for the double quotes?

Reply:
You could also consider putting all of this together into the larger `sh` invocation above, which would also obviate the need to think about escapes.